### ISSUE TITLE
    ValueError: Unknown type <class 'redis.client.StrictPipeline'>

### ISSUE THREAD

**Issue by kc1** [rq/rq#938]
[![enter image description here][1]][1]

I develop locally on win10, which is a problem for the usage of the RQ task queue, which only works on linux systems because it requires the ability to fork processes. I'm trying to extend the flask-base project https://github.com/hack4impact/flask-base/tree/master/app which can use RQ. I came across https://github.com/michaelbrooks/rq-win . I love the idea of this repo (If I can get it working it will really simplify my life, since I develop on win 10 -64):

After installing this library 

I can queue a job in my views by running something like:

         @login_required
         @main.route('/selected')
         def selected():
         messages = 'abcde'
          j = get_queue().enqueue(render_png, messages, result_ttl=5000)
          return j.get_id()

This returns a job_code correctly. 

I changed the code in manage.py to:

    from rq_win import WindowsWorker

    @manager.command
    def run_worker():
        """Initializes a slim rq task queue."""
        listen = ['default']

        REDIS_URL = 'redis://localhost:6379'
        conn = Redis.from_url(REDIS_URL)

        with Connection(conn):
            # worker = Worker(map(Queue, listen))
            worker = WindowsWorker(map(Queue, listen))
            worker.work()
When I try to run it with:

    $ python -u manage.py run_worker 

    09:40:44
    09:40:44 *** Listening on ?[32mdefault?[39;49;00m...
    09:40:58 ?[32mdefault?[39;49;00m: ?[34mapp.main.views.render_png('{"abcde"}')?[39;49;00m (8c1b6186-39a5-4daf-9c45-f60e4241cd1f)
    ...\lib\site-packages\rq\job.py:161: DeprecationWarning: job.status is deprecated. Use job.set_status() instead
      DeprecationWarning
    09:40:58 ?[31mValueError: Unknown type <class 'redis.client.StrictPipeline'>?[39;49;00m
    Traceback (most recent call last):
      File "...\lib\site-packages\rq_win\worker.py", line 87, in perform_job
        queue.enqueue_dependents(job, pipeline=pipeline)
      File "...\lib\site-packages\rq\queue.py", line 322, in enqueue_dependents
        for job_id in pipe.smembers(dependents_key)]
      File "...\lib\site-packages\rq\queue.py", line 322, in <listcomp>
        for job_id in pipe.smembers(dependents_key)]
      File "...\lib\site-packages\rq\compat\__init__.py", line 62, in as_text
        raise ValueError('Unknown type %r' % type(v))
    ValueError: Unknown type <class 'redis.client.StrictPipeline'>

So in summary, I think the jobs are being queued correctly within redis. However when the worker process tries to grab a job off of the queue to process, This error occurs. How can I fix this?

  [1]: https://i.stack.imgur.com/DDZGc.png

**Comment by selwin** [comment]
RQ does not officially support Windows. However, if the fix is simple and does not add significant complexity to RQ's code base, please submit a PR and I look into merging it.

This PR may also be related: https://github.com/rq/rq/pull/860

**Comment by OlegZv** [comment]
@selwin I'd like to reopen this as I think it's an actual issue. I'm not sure why this doesn't happen currently...
This part of the code, if `Pipeline` is passed into the function, will not return the job_ids, but rather an instance of the `Pipeline`.
https://github.com/rq/rq/blob/7b0d622ce35265acac32291e5350ed394b013dc4/rq/queue.py#L1220

**Comment by selwin** [comment]
I reopened this issue. Can you provide a test case demonstrating this issue?

**Comment by OlegZv** [comment]
Yes. I have multiple test cases I'm working on. I'm sorry for the delay, but when I'm introducing a change I want to make sure it's backward compatible and has plenty of unit-tests. I should have some PRs coming your way this week (with tests :) )
